src/app/global-vars.service.ts

```diff
@@ -156,35 +157,87 @@ export class GlobalVarsService {
 
   amplitude: AmplitudeClient;
 
-  _setNumMessagesToRead() {
-    if (
-      !this.loggedInUser ||
-      !this.messageResponse ||
-      !this.messageResponse.OrderedContactsWithMessages ||
-      !this.messageResponse.TotalMessagesByContact ||
-      !this.messageResponse.MessageReadStateByContact
-    ) {
+  _setupMessages() {
+    // If there's no loggedInUser, we set the notification count to zero
+    if (!this.loggedInUser) {
       this.messageNotificationCount = 0;
       return;
     }
-    let totalMessages = 0;
-    let totalRead = 0;
-    for (let contact of this.messageResponse.OrderedContactsWithMessages) {
-      if (contact.Messages.length > 0) {
-        if (this.messageResponse.TotalMessagesByContact[contact.PublicKeyBase58Check]) {
-          totalMessages += this.messageResponse.TotalMessagesByContact[contact.PublicKeyBase58Check];
-        }
-        if (this.messageResponse.MessageReadStateByContact[contact.PublicKeyBase58Check]) {
-          totalRead += this.messageResponse.MessageReadStateByContact[contact.PublicKeyBase58Check];
-        }
-      }
+
+    // If a message response already exists, we skip this step
+    if (this.messageResponse) {
+      return;
     }
-    let notificationCount = totalMessages - totalRead;
-    if (notificationCount >= 0) {
-      this.messageNotificationCount = notificationCount;
-    } else {
-      this.messageNotificationCount = 0;
+
+    let storedTab = this.backendApi.GetStorage("mostRecentMessagesTab");
+    if (storedTab === null) {
+      storedTab = "My Holders";
+      this.backendApi.SetStorage("mostRecentMessagesTab", storedTab);
+    }
+
+    // Set the filters most recently used and load the messages
+    this._setMessagesFilter(storedTab);
+    this._loadInitialMessages();
+  }
+
+  _setMessagesFilter(tabName: any) {
```

Comment:
by convention, function names that begin with an underscore are considered private.  Since this function is used outside of this component, let's remove the underscore.

Reply:
note: we haven't been great about this, but would like to try to be more consistent going forward.